### ISSUE TITLE
`verifying` throws NPE where an equivalent `providing` works

### ISSUE THREAD

**Issue by FelipeCortez** [nubank/mockfn#4]
With `deps.edn`:

```clojure
{:deps {nubank/matcher-combinators {:mvn/version "1.2.4"}
        mockfn                     {:mvn/version "0.5.0"}}}
```

```clojure
(require '[clojure.test :refer [is]]
         '[matcher-combinators.standalone :refer [match?]]
         '[matcher-combinators.matchers :as m]
         '[mockfn.macros :refer [providing verifying]]
         '[mockfn.matchers :as mfn.match])

(def a-fn (constantly :a))

(providing [(a-fn (match? (m/in-any-order [1 2]))) :b]
  (is (= :b (a-fn [2 1]))))
;; => true

(verifying [(a-fn (match? (m/in-any-order [1 2]))) :b (mfn.match/exactly 1)]
  (is (= :b (a-fn [2 1]))))
;; => Execution error (NullPointerException) at mockfn.mock/verify (mock.cljc:21).
;;    null
```

**Comment by dchelimsky** [comment]
Here's a simpler failure case:

```clojure
(verifying [(a-fn (match? 1)) :b (mfn.match/exactly 1)]
           (is (= :b (a-fn 1))))
```

The problem has to do with using `(match? <args>)` as a key in more than one map, but re-evaluating it for each map. This works fine with vars, but not anonymous functions e.g.:

```clojure
(defn f [] (fn []))

(= (f) (f))
;; => false
```

**Comment by dchelimsky** [comment]
Here's the function where it happens: https://github.com/nubank/mockfn/blob/e364a0f4f8503a114b840ab36b330fb398e44adc/src/mockfn/mock.cljc#L17-L23

**Comment by gabrielgiussi** [comment]
The thing is that for anonymous functions we are creating one instance of it for each entry in the map (i.e. `:return-values`, `:times-called` and `:times-expected`) so then when we look up for `times-called` using the keys from `:times-expected`
```clojure
(let [times-called (-> spec :times-called (get args) deref)] 
```
we are getting nil and NPE.

If we macroexpand the verifying we can easily see that we are creating one instance for each entry
```clojure
(macroexpand '(macros/verifying [(fixtures/one-fn (fn [x] (= 1 x))) :a (matchers/exactly 1)] 
                                  (is (= :a (fixtures/one-fn 1)))))


(clojure.core/with-redefs-fn
 {(var fixtures/one-fn) (mockfn.mock/mock
                         fixtures/one-fn
                         {:function fixtures/one-fn,
                          :return-values {[(fn [x] (= 1 x))] :a},
                          :times-called {[(fn [x] (= 1 x))] (clojure.core/atom 0)},
                          :times-expected {[(fn [x] (= 1 x))] [(matchers/exactly 1)]}})}
 (clojure.core/fn
  []
  (clojure.core/let
   [res__3556__auto__ (do (is (= :a (fixtures/one-fn 1))))]
   (clojure.core/doseq
    [mock__3557__auto__
     (clojure.core/keys
      #:fixtures{one-fn {:function fixtures/one-fn,
                         :return-values {[(fn [x] (= 1 x))] :a},
                         :times-called {[(fn [x] (= 1 x))] (clojure.core/atom 0)},
                         :times-expected {[(fn [x] (= 1 x))] [(matchers/exactly 1)]}}})]
    (mockfn.mock/verify mock__3557__auto__))
   res__3556__auto__)))
```

The function `(fn [x] (= 1 x))` appears as a list and not as a function.

**Comment by dchelimsky** [comment]
Fixed in #5 
Released in 0.6.1
